cli/src/main/java/com/devonfw/tools/ide/environment/EnvironmentVariablesPropertiesFile.java

```diff
@@ -46,29 +49,80 @@ public final class EnvironmentVariablesPropertiesFile extends EnvironmentVariabl
   public EnvironmentVariablesPropertiesFile(AbstractEnvironmentVariables parent, EnvironmentVariablesType type,
       Path propertiesFilePath, IdeContext context) {
 
+    this(parent, type, getParent(propertiesFilePath), propertiesFilePath, context);
+  }
+
+  /**
+   * The constructor.
+   *
+   * @param parent the parent {@link EnvironmentVariables} to inherit from.
+   * @param type the {@link #getType() type}.
+   * @param propertiesFolderPath the {@link Path} to the folder where the properties file is expected.
+   * @param propertiesFilePath the {@link #getSource() source}.
+   * @param context the {@link IdeContext}.
+   */
+  public EnvironmentVariablesPropertiesFile(AbstractEnvironmentVariables parent, EnvironmentVariablesType type, Path propertiesFolderPath,
+      Path propertiesFilePath, IdeContext context) {
+
     super(parent, context);
     Objects.requireNonNull(type);
     assert (type != EnvironmentVariablesType.RESOLVED);
     this.type = type;
-    this.propertiesFilePath = propertiesFilePath;
+    if (propertiesFolderPath == null) {
+      this.propertiesFilePath = null;
+      this.legacyPropertiesFilePath = null;
+    } else {
+      if (propertiesFilePath == null) {
+        this.propertiesFilePath = propertiesFolderPath.resolve(DEFAULT_PROPERTIES);
+      } else {
+        this.propertiesFilePath = propertiesFilePath;
+        assert (propertiesFilePath.getParent().equals(propertiesFolderPath));
+      }
+      Path legacyPropertiesFolderPath = propertiesFolderPath;
+      if (type == EnvironmentVariablesType.USER) {
+        // ~/devon.properties vs. ~/.ide/ide.properties
+        legacyPropertiesFolderPath = propertiesFolderPath.getParent();
+      }
+      this.legacyPropertiesFilePath = legacyPropertiesFolderPath.resolve(LEGACY_PROPERTIES);
+    }
```

Comment:
Maybe move into another method to distinct its use case

Reply:
If you could do that please go ahead.

Reply:
Nothing to do here, resolved.